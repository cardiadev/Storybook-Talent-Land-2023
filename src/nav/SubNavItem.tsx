import React from "react";
import classNames from "classnames";

import { Typography } from "../Typography";
import { ISubNavItem } from "../@interfaces";

interface ISubNavItemProps {
  inDropdown?: boolean;
  isActive: boolean;
  subItem: ISubNavItem;
  onClick: React.MouseEventHandler<HTMLLIElement>;
}

export const SubNavItem = ({
  inDropdown = false,
  isActive,
  subItem,
  onClick,
}: ISubNavItemProps) => (
  <li
    key={subItem.label}
    className={classNames("navItem group", {
      "bg-primary-50 dark:bg-gray-100 dark:bg-opacity-10": isActive,
    })}
    onClick={onClick}
  >
    <Typography
      variant="md"
      customWeight="medium"
      className={classNames("flex-grow", {
        "pl-13": !inDropdown,
        "pl-4": inDropdown,
      })}
      customColor={classNames(
        "group-hover:text-primary-600 dark:group-hover:text-white",
        {
          "text-gray-800 dark:text-white": !isActive,
          "text-primary-600 dark:text-white": isActive,
        },
      )}
    >
      {subItem.label}
    </Typography>
  </li>
);
