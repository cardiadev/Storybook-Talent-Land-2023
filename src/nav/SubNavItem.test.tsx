import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { SubNavItem } from "./SubNavItem";

afterEach(cleanup);

const setup = () => {
  const subItems = [
    {
      order: 0,
      path: "analytics",
      label: "Analytics",
    },
    {
      order: 1,
      path: "apps",
      label: "Apps",
    },
    {
      order: 2,
      path: "learn",
      label: "Learn",
    },
  ];
  const activeSubNavItem = "all";

  return {
    subItems,
    activeSubNavItem,
  };
};

describe("SubNavItem", () => {
  it("Renders SubNavItems correctly", () => {
    const { subItems, activeSubNavItem } = setup();

    const { asFragment } = render(
      <>
        {subItems.map((subItem) => (
          <SubNavItem
            key={subItem.label}
            isActive={activeSubNavItem === subItem.label}
            subItem={subItem}
            onClick={() => {}}
          />
        ))}
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders SubNavItems with inDropdown=true correctly", () => {
    const { subItems, activeSubNavItem } = setup();

    const { asFragment } = render(
      <>
        {subItems.map((subItem) => (
          <SubNavItem
            key={subItem.label}
            inDropdown
            isActive={activeSubNavItem === subItem.label}
            subItem={subItem}
            onClick={() => {}}
          />
        ))}
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
