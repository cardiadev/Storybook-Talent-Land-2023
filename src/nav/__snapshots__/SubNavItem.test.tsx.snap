// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNavItem Renders SubNavItems correctly 1`] = `
<DocumentFragment>
  <li
    class="navItem group"
  >
    <p
      class="text-md font-medium flex-grow pl-13 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
    >
      Analytics
    </p>
  </li>
  <li
    class="navItem group"
  >
    <p
      class="text-md font-medium flex-grow pl-13 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
    >
      Apps
    </p>
  </li>
  <li
    class="navItem group"
  >
    <p
      class="text-md font-medium flex-grow pl-13 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
    >
      Learn
    </p>
  </li>
</DocumentFragment>
`;

exports[`SubNavItem Renders SubNavItems with inDropdown=true correctly 1`] = `
<DocumentFragment>
  <li
    class="navItem group"
  >
    <p
      class="text-md font-medium flex-grow pl-4 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
    >
      Analytics
    </p>
  </li>
  <li
    class="navItem group"
  >
    <p
      class="text-md font-medium flex-grow pl-4 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
    >
      Apps
    </p>
  </li>
  <li
    class="navItem group"
  >
    <p
      class="text-md font-medium flex-grow pl-4 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
    >
      Learn
    </p>
  </li>
</DocumentFragment>
`;
